File: 2. Java Advanced/1. BigInteger/4. BigIntegerPrime/Prime.java
import java.math.BigInteger;
import java.util.Scanner;
public class Prime{
    public static void main(String[] args){
        
        Scanner input = new Scanner(System.in);
        System.out.print("Enter a number: ");
        int num = input.nextInt();
        System.out.println(num+" is Prime "+isPrime(num));
        System.out.println("next Prime after "+num+" is "+nextPrime(num));
        input.close();
    }

    public static boolean isPrime(int num){
        BigInteger a = BigInteger.valueOf(num);
        return a.isProbablePrime(1);
    }

    public static BigInteger nextPrime(int num){
        return BigInteger.valueOf(num).nextProbablePrime();
    }
}
//0,1,1,2,3,5,8,13,21

File: Python/Math/Prime.py
# T(n): O(n)
# S(n): O(1)
def isPrime(num):
   if num <= 1: return False
   for i in range(2, num):
      if num % i == 0:
         return False
   return True


# T(n): O(sqrt(n))
# S(n): O(1)
import math
def isPrime2(num):
  if num <= 1: return False
  for i in range(2, int(math.sqrt(num))):
    if (num % i == 0):
      return False
  return True


print(isPrime(7000))
print(isPrime2(7000))